libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/theme/components/ListItem.kt

```diff
@@ -0,0 +1,419 @@
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import io.element.android.libraries.designsystem.preview.ElementThemedPreview
+import io.element.android.libraries.designsystem.preview.PreviewGroup
+import io.element.android.libraries.theme.ElementTheme
+
+// Designs: https://www.figma.com/file/G1xy0HDZKJf5TCRFmKb5d5/Compound-Android-Components?type=design&node-id=425%3A24208&mode=design&t=G5hCfkLB6GgXDuWe-1
+
+/**
+ * A List Item component to be used in lists and menus with simple layouts, matching the Material 3 guidelines.
+ * @param headlineContent The main content of the list item, usually a text.
+ * @param modifier The modifier to be applied to the list item.
+ * @param overlineContent The content to be displayed above the headline content.
+ * @param supportingContent The content to be displayed below the headline content.
+ * @param leadingContent The content to be displayed before the headline content.
+ * @param trailingContent The content to be displayed after the headline content.
+ * @param colors The colors to be used for the list item.
+ * @param tonalElevation The tonal elevation to be used for the list item.
+ * @param shadowElevation The shadow elevation to be used for the list item.
+ * @param isPrimaryAction Whether the list item is a primary action. This will change the color of the leading content to use primary tokens.
+ * @param isDestructive Whether the list item is a destructive action. This will change the color of the headline content to use critical tokens.
+ * @param enabled Whether the list item is enabled. When disabled, will change the color of the headline content and the leading content to use disabled tokens.
+ * @param onClick The callback to be called when the list item is clicked.
+ *
+ * **Note:** [isPrimaryAction], [isDestructive] and [enabled] should only be used in single line [ListItem] components,
+ * and only one should be customised at a time.
+ */
+@Suppress("LongParameterList")
+@Composable
+fun ListItem(
+    headlineContent: @Composable () -> Unit,
+    modifier: Modifier = Modifier,
+    overlineContent: @Composable (() -> Unit)? = null,
+    supportingContent: @Composable (() -> Unit)? = null,
+    leadingContent: @Composable (() -> Unit)? = null,
+    trailingContent: @Composable (() -> Unit)? = null,
+    colors: ListItemColors = ListItemDefaults.colors(),
+    tonalElevation: Dp = ListItemDefaults.Elevation,
+    shadowElevation: Dp = ListItemDefaults.Elevation,
+    isPrimaryAction: Boolean = false,
+    isDestructive: Boolean = false,
+    enabled: Boolean = true,
+    onClick: () -> Unit = { },
```

Comment:
There is always a pb with having a default empty lambda, which is when the lambda stays empty, the click effect is still visible (since we invoke `modifier.clickable(enabled = enabled, onClick = onClick)` below). I think we may have `ListItem` with no click action, but if not, maybe remove the default parameter value. WDYT?

Reply:
Done in https://github.com/vector-im/element-x-android/pull/990/commits/53ebf11b6a791e315bac375a9f24bb92f22e5b04.